foojay-resolver/src/main/kotlin/org/gradle/toolchains/foojay/packages.kt

```diff
@@ -32,13 +39,18 @@ fun OperatingSystem.toApiValue(): String =
         OperatingSystem.MAC_OS -> "macos"
         OperatingSystem.SOLARIS -> "solaris"
         OperatingSystem.FREE_BSD -> "linux"
+        OperatingSystem.AIX -> "aix"
     }
 
 private fun matches(p: Package, architecture: Architecture): Boolean =
     when (architecture) {
         Architecture.X86 -> p.architecture in architectures32Bit
         Architecture.X86_64 -> p.architecture in architectures64Bit
         Architecture.AARCH64 -> p.architecture in architecturesArm64Bit
+        Architecture.PPC64 -> p.architecture in architecturesPpc64Bit
```

Comment:
I would consider to not introduce a dependency to latest Gradle version in `main` code but rather detect newly introduced architectures based on `architecture.name()`. This way the exact same plugin can be used for old and new Gradle versions.

Reply:
> I would consider to not introduce a dependency to latest Gradle version in `main` code but rather detect newly introduced architectures based on `architecture.name()`. This way the exact same plugin can be used for old and new Gradle versions.

Actually doesn't seem to be the case, there is no issue with older Gradle versions (functional tests are cross version now and they prove it).